src/VisualStudio/Core/Def/Implementation/Diagnostics/MiscellaneousDiagnosticAnalyzerService.cs

```diff
@@ -82,7 +90,9 @@ public SyntaxOnlyDiagnosticAnalyzer(MiscellaneousDiagnosticAnalyzerService servi
                 Contract.Requires(document.Project.Solution.Workspace == _workspace);
 
                 var diagnosticData = diagnostics == null ? ImmutableArray<DiagnosticData>.Empty : diagnostics.Select(d => DiagnosticData.Create(document, d)).ToImmutableArrayOrEmpty();
-                _service.RaiseDiagnosticsUpdated(new DiagnosticsUpdatedArgs(new MiscUpdateArgsId(document.Id), _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
+                _service.RaiseDiagnosticsUpdated(
+                    new DiagnosticsUpdatedArgs(new MiscUpdateArgsId(document.Id, _service.analyzerService.GetCompilerDiagnosticAnalyzer(document.Project.Language)),
+                    _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
```

Comment:
Looks like this will cause the source to show up as "VisualBasicCompilerDiagnosticAnalyzer [Microsoft.CodeAnalysis]". Should we display this via some well known string like "VB Compiler Live Analysis" instead? I think this can be done via creating a CompilerLiveUpdateArgsId type derived from LiveUpdateArgsId (which always returns a fixed string). And if we do this, we probably won't need the GetCompilerDiagnosticAnalyzer() API at all...